src/config_map.py

```diff
@@ -94,7 +94,10 @@ def pop(self, keys: List[str]):
             return
 
         for key in keys:
-            cm.data.pop(key)
+            if key in cm.data.keys():
```

Comment:
minor: If the logging below is needed, we could use `try` and `except` block instead of using`if` to see if the kv is there.
If the logging is not necessary, we could do something like: `cm.data.pop(key, None)` to avoid the possible `KeyError`.